libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1230,6 +1246,28 @@ string YulUtilFunctions::resizeDynamicByteArrayFunction(ArrayType const& _type)
 	});
 }
 
+string YulUtilFunctions::cleanUpDynamicByteArrayEndSlotsFunction(ArrayType const& _type)
+{
+	solAssert(_type.isByteArray(), "");
+	solAssert(_type.isDynamicallySized(), "");
+
+	string functionName = "clean_up_byte_array_end_slots_" + _type.identifier();
+	return m_functionCollector.createFunction(functionName, [&](vector<string>& _args, vector<string>&) {
+		_args = {"slot", "len", "pos"};
```

Comment:
Nit: it is called slot here, yet the documentation names it array.

Reply:
Fixed